lib/config.js

```diff
@@ -66,13 +66,14 @@ class Config {
         this.parser = options.parser;
         this.parserOptions = options.parserOptions || {};
 
+        this.configCache = new ConfigCache();
```

Comment:
I'm not sure. Personally I think it would make more sense to omit `extends` entirely after the config has already been merged, but that seems like a separate issue.